File: standard-java/src/main/java/com/jetbrains/refactoring/ExtractParameter.java
package com.jetbrains.refactoring;

import java.lang.reflect.Parameter;


public class ExtractParameter {
    public void extractParameter(Parameter p) {
        //Extract parameter via Ctrl/Cmd+Alt+P
        Parameter parameter = getClass().getEnclosingMethod().getParameters()[0];

        System.out.println(parameter.getName());
    }
}

File: kotlin/src/main/java/com/jetbrains/refactoring/ExtractParameter.kt
package com.jetbrains.refactoring.kotlin

@Suppress("unused", "SameParameterValue")
class ExtractParameter {

    fun callsGreetPerson() {
        greetPerson("you")
    }

    private fun greetPerson(personName: String) {
        // Extract parameter: Ctrl+Alt+P/Cmd+Option+P
        // Try different options, such as setting a default value
        val greeting = "Hello, "
        println(greeting + personName)
    }

}